src/components/screens/editor/editor-screen.tsx

```diff
@@ -1,12 +1,273 @@
+import { ListRounded, KeyboardArrowDown, Settings, DateRangeRounded, PersonRounded, Add } from "@mui/icons-material";
+import { useApiClient } from "app/hooks";
+import { ErrorContext } from "components/contexts/error-handler";
 import strings from "localization/strings";
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import Api from "api";
+import { NavigationTabContainer } from "styled/layouts/navigations";
+import NavigationTab from "components/layouts/navigations/navigation-tab";
+import { Metaform, MetaformVersion, MetaformVersionType } from "generated/client";
+import { AdminFormListStack, AdminFormTypographyField } from "styled/react-components/react-components";
+import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import { NewUserButton } from "styled/layouts/admin-layout";
+import moment from "moment";
+import EditorScreenDrawer from "./editor-screen-drawer";
+import { useNavigate } from "react-router-dom";
+
+/**
+ * Interface for single Metaform row
+ */
+interface Row {
+  metaform: Metaform;
+  versions: MetaformVersion[];
+}
 
 /**
  * Editor screen component
  */
-const EditorScreen: React.FC = () => (
-  <Typography>{ strings.generic.notImplemented }</Typography>
-);
+const EditorScreen: React.FC = () => {
+  const errorContext = useContext(ErrorContext);
+  const navigate = useNavigate();
+
+  const apiClient = useApiClient(Api.getApiClient);
+  const { metaformsApi, versionsApi } = apiClient;
+
+  const [ metaforms, setMetaforms ] = useState<Row[]>([]);
+  const [ loading, setLoading ] = useState<boolean>(false);
+
+  const [ drawerOpen, setDrawerOpen ] = useState<boolean>(false);
+
+  /**
+   * Gets Metaforms and corresponding MetaformVersions 
+   */
+  const loadMetaforms = async () => {
+    setLoading(true);
+    
+    try {
+      const forms = await metaformsApi.listMetaforms({});
+      const rows: Row[] = await Promise.all(forms.map(async form => {
+        return {
+          metaform: form,
+          versions: await versionsApi.listMetaformVersions({ metaformId: form.id!! })
+        };
+      }));
+
+      setMetaforms(rows);
+    } catch (e: any) {
+      errorContext.setError("Error while listing forms.");
+    }
+
+    setLoading(false);
+  };
+
+  /**
+   * Creates new Metaform and MetaformVersion
+   */
+  const createMetaform = async (metaform: Metaform) => {
+    try {
+      const newMetaform = await metaformsApi.createMetaform({ metaform: metaform });
+      const newMetaformVersion = await versionsApi.createMetaformVersion({
+        metaformId: newMetaform.id!!,
+        metaformVersion: {
+          type: MetaformVersionType.Draft,
+          data: { ...newMetaform } as any
+        }
+      });
+      const currentPath = window.location.pathname;
+      navigate(`${currentPath}/${newMetaform.slug}/${newMetaformVersion.id}`);
+    } catch (e: any) {
+      errorContext.setError("Error while creating new Metaform.");
+    }
```

Comment:
Use localized error message and pass actual error also to error handler, no need to type error as any.